src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1289,6 +1295,12 @@ .. chosenResolvedItem.Suppressions
                     LibraryDependency childDependency = chosenResolvedItem.Item.Data.Dependencies[i];
                     LibraryDependencyIndex childLibraryDependencyIndex = chosenResolvedItem.GetDependencyIndexForDependencyAt(i);
 
+                    if (childLibraryDependencyIndex == LibraryDependencyIndex.Project)
+                    {
+                        // Skip any dependency with the same name as the project
```

Comment:
Does it need to guard against `LibraryDependencyIndex.Invalid` ?

Reply:
Great point, since the `DependencyGraphItem.LibraryDependencyIndex` property has the `init` modifier set, there's probably no need for the `Invalid` enum value.  I'll remove since its not needed.  This should mean that the value is either Project or a positive integer greater than `0`.